_posts/2019-03-18-legacy-rails-apps-silently-judging-you.md

```diff
@@ -0,0 +1,236 @@
+[`rails-erd`](https://github.com/voormedia/rails-erd) to quickly generate an
+Entity Relationship Diagram.
+
+## Static Code Analysis/Judgment
+
+You can use tools like `flay`, `flog`, and `heckle` to do static code analysis
+and get a sense of what parts of the application will cause most of
+the maintenance pain. More info over here: [Ruby Sadist](http://ruby.sadi.st/Ruby_Sadist.html) (NSFW)
+
+We prefer [CodeClimate](https://codeclimate.com) for this. You can get a quick
+sense of the maintainability level of your application:
+
+<img src="/blog/assets/images/code-climate-report.png" alt="Sample Code Climate Report">
+
+You can filter all the existing issues to get a sense of the most critical
+issues:
+
+<img src="/blog/assets/images/code-climate-report.png" alt="Sample Code Climate Report">
+
+Code Climate will give you an idea of the complexity of the application.
+
+There are certainly many free, open source alternatives, which you can find over
+here: [https://github.com/metricfu/metric_fu/wiki/Code-Tools](https://github.com/metricfu/metric_fu/wiki/Code-Tools)
+
+## Developer Onboarding Judgment
+
+Does the application provide a well-maintained `Dockerfile`? Does it provide a
+`./bin/setup` script? Do they run?
+
+Setting up a new development environment should be as simple as:
+
+```
+./bin/setup`
+```
+
+Or:
+
+```
+docker-compose up
+```
+
+Or:
+
+```
+vagrant up
+```
+
+You **get** the point. It is a **red flag** when you need to pair with someone to get
+your development environment set up. You should count the times you run into
+unexpected issues. This will give you a good idea of the complexity of *all*
+environments.
+
+## Build Judgment
+
+Is there a set of spec files? Does the build work in your environment? After you
+have set up your local environment, you should be able to run:
+
+```
+rake
+```
+
+I'm used to projects and gems using `test` or `spec` as the default Rake task,
+which has been the default task for Rails for a while: [https://github.com/rails/rails/blob/5-2-stable/Rakefile#L21-L22](https://github.com/rails/rails/blob/5-2-stable/Rakefile#L21-L22)
+
+```ruby
+# Rakefile
+desc "Run all tests by default"
+task default: %w(test test:isolated)
+```
+
+It is a **red flag** when you run into unexpected issues when running the build.
+If some tests failed, why did they fail? How many random failures did you get? If
+you found many issues, why didn't `./bin/setup` take care of it for you beforehand?
+
+## Coverage Judgment
+
+Let's say your application is so well maintained that all previous steps worked
+flawlessly. `bundle outdated` returned no outdated dependencies; static code
+analysis reported an A+; you run `./bin/setup` to install everything you needed;
+and the `rake` showed you a passing build. That is great!
+
+How well covered is it? Does it cover more than 80% of your code? Does it cover
+100% of the [critical path]()?
+
+You can quickly find out with `SimpleCov`. Simply add this to your `Gemfile`:
```

Comment:
@etagwerker Maybe you can include a link to https://github.com/colszowka/simplecov